### PR TITLE
Work

### PR DESCRIPTION
**Pull request by Leroys2019**

### CODE REVIEW

File: src/main/java/Problem3/Book.java [hunk 1]
```diff
@@ -16,10 +16,20 @@ public Book(String title, String author) {
...
+        this.author = anotherBook.author;
+        this.id = anotherBook.id;
+
     }
 
     @Override
     public boolean equals(Object obj) {
         // homework
+        Book b = (Book) obj;
+        if ((b.title == this.title) && (b.id == this.id) && (b.author == this.author)) {
```
Comment (pdgetrf) [inline src/main/java/Problem3/Book.java:29]: wrong logic 

```
A movie equals another movie only when their id are the same. Same rule for books being equal.
```

File: src/main/java/Problem3/Book.java [hunk 2]
```diff
@@ -16,10 +16,20 @@ public Book(String title, String author) {
...
+            return true;
+        } else {
```
Comment (pdgetrf) [inline src/main/java/Problem3/Book.java:31]: "else" not needed here

File: src/main/java/Problem3/BookFiction.java [hunk 3]
```diff
@@ -9,12 +9,14 @@
     public BookFiction(String title, String author, String genres) {
         // homework
         // tip: use the 'super' keyword
+        super(title, author);
```
Comment (pdgetrf) [inline src/main/java/Problem3/BookFiction.java:12]: missing genres

File: src/main/java/Problem3/BookFiction.java [hunk 4]
```diff
@@ -9,12 +9,14 @@
...
     }
 
     // copy constructor
     public BookFiction(BookFiction anotherBook) {
         // homework
         // tip: use the 'super' keyword
+        super(anotherBook);
```
Comment (pdgetrf) [inline src/main/java/Problem3/BookFiction.java:19]: missing genres

File: src/main/java/Problem3/BookFiction.java [hunk 5]
```diff
@@ -25,5 +27,11 @@ public int getLateFeeInDollar() {
     @Override
     public int calcLateFees(int numOfDaysPastDue) {
         // homework
+        if (numOfDaysPastDue > 0 && numOfDaysPastDue < 5) {
+            return(numOfDaysPastDue * lateFeePerDayInDollar);
+        } else if (numOfDaysPastDue >= 5) {
+            return((numOfDaysPastDue * lateFeePerDayInDollar));
```
Comment (pdgetrf) [inline src/main/java/Problem3/BookFiction.java:33]: is it just me or both cases are returning the same value? can you make it one case?

File: src/main/java/Problem3/BookRomance.java [hunk 6]
```diff
@@ -22,5 +24,11 @@ public int getLateFeeInDollar() {
     @Override
     public int calcLateFees(int numOfDaysPastDue) {
         // homework
+        if (numOfDaysPastDue > 0 && numOfDaysPastDue < 5) {
```
Comment (pdgetrf) [inline src/main/java/Problem3/BookRomance.java:27]: same here.

File: src/main/java/Problem3/Movie.java [hunk 7]
```diff
@@ -8,16 +8,28 @@
...
+        this.rating = anotherMovie.rating;
+        this.id = anotherMovie.id;
     }
 
     @Override
     public boolean equals(Object obj) {
-        // homework
+        //homework
+        Movie m = (Movie) obj;
+        if ((m.title == this.title) && (m.id == this.id) && (m.rating == this.rating)) {
```
Comment (pdgetrf) [inline src/main/java/Problem3/Movie.java:29]: wrong logic again here.

File: src/main/java/Problem3/MovieAction.java [hunk 8]
```diff
@@ -22,5 +25,12 @@ public int getLateFeeInDollar() {
     @Override
     public int calcLateFees(int numOfDaysPastDue) {
         // homework
+        if (numOfDaysPastDue > 0 && numOfDaysPastDue < 5) {
+            return(numOfDaysPastDue * lateFeePerDayInDollar);
+        } else if (numOfDaysPastDue >= 5) {
+            return(2 * (numOfDaysPastDue * lateFeePerDayInDollar));
```
Comment (pdgetrf) [inline src/main/java/Problem3/MovieAction.java:31]: duplicated logic here. 

suggesting:
```java
        int fee = lateFeePerDayInDollar * numOfDaysPastDue;
        return (numOfDaysPastDue >= 5) ? 2 * fee : fee;
```

and here number 5 only appears once too

File: src/main/java/Problem3/MovieComedy.java [hunk 9]
```diff
@@ -21,5 +23,13 @@ public int getLateFeeInDollar() {
     @Override
     public int calcLateFees(int numOfDaysPastDue) {
         // homework
+        if (numOfDaysPastDue > 0 && numOfDaysPastDue < 5) {
```
Comment (pdgetrf) [inline src/main/java/Problem3/MovieComedy.java:26]: same here

File: src/test/java/Problem2Test.java [hunk 10]
```diff
@@ -7,13 +7,17 @@
...
+                {0, 1, 1, 1, 1, 1, 1, 2}
         };
 
         int[][] expects = {
-                {0, 1, 2, 4, 6}
-                // add more test cases
+                {0, 1, 2, 4, 6},
+                {0, 1, 2, 6},
+                {0, 1, 2, 3, 4, 5, 6},
+                {0, 1, 2}
```
Comment (pdgetrf) [inline src/test/java/Problem2Test.java:20]: great test. let's add more cases for the edge cases, such as null array and empty array.

**Review by pdgetrf — APPROVED** [review]
